pontoon/tags/utils/base.py

```diff
@@ -0,0 +1,193 @@
+
+from collections import OrderedDict
+
+from django.db.models import Q
+from django.utils.functional import cached_property
+
+from pontoon.base.models import (
+    Locale, Project, Resource, TranslatedResource, Translation)
+from pontoon.base.utils import glob_to_regex
+from pontoon.tags.models import Tag
+
+
+class Clonable(object):
+
+    """Instantiated descendants of this class can be called to create a cloned
+    version of the object.
+
+    The clone will be called with attributes listed in `self.clone_kwargs` as
+    kwargs. These can be overridden when creating the clone.
+
+    """
+    clone_kwargs = ()
+
+    def __init__(self, **kwargs):
+        for k in self.clone_kwargs:
+            setattr(self, k, kwargs.get(k))
+
+    def __call__(self, **kwargs):
+        clone_kwargs = dict(
+            (k, getattr(self, k))
+            for k
+            in self.clone_kwargs)
+        clone_kwargs.update(kwargs)
+        return self.__class__(**clone_kwargs)
+
+
+class FilteredData(Clonable):
+    """Base Tool for constructing and coalescing aggregating querysets
```

Comment:
Nit: Given that the comments says it's a tool, should the class name be `FilteredDataTool`, to resemble the naming schema used in descendants?

Reply:
sounds good - ill update